tests/test_messagers.py

```diff
@@ -0,0 +1,505 @@
+            }
+
+    conn = boto3.resource("s3")
+    client = boto3.client("s3")
+    conn.create_bucket(Bucket="testbucket")
+    conn.create_bucket(Bucket="testbucket2")
+    client.put_object(Bucket="testbucket", Key="testprefix/k4", Body="unset")
+    client.put_object(Bucket="testbucket", Key="testprefix/k5", Body="unset")
+
+    producer = Mock()
+
+    testmessager = TestMessager(client, "testbucket", "testprefix/", producer)
+    assert testmessager.consume("") == Messager.Failures(permanent=False, temporary=False)
+
+    obj1 = client.get_object(Bucket="testbucket2", Key="testprefix/k1")
+    assert obj1["ContentType"] == "application/json"
+    assert obj1["Body"].read() == b"test_body1"
+
+    obj1 = client.get_object(Bucket="testbucket", Key="testprefix/k2")
+    assert obj1["ContentType"] == "application/json"
+    assert obj1["Body"].read() == b"test_body2"
+
+    obj1 = client.get_object(Bucket="testbucket", Key="testprefix/k3")
+    assert obj1["ContentType"] == "x-test"
+    assert obj1["Body"].read() == b"test_body3"
+
+    obj1 = client.get_object(Bucket="testbucket", Key="testprefix/k4")
+    assert obj1["ContentType"] == "application/json"
+    assert obj1["Body"].read() == b"test_body4"
+
+    try:
+        obj1 = client.get_object(Bucket="testbucket", Key="testprefix/k5")
+    except botocore.exceptions.ClientError as e:
+        assert e.response["Error"]["Code"] == "NoSuchKey" or e.response["Error"]["Code"] == "404"
+
+    message = producer.send.call_args.args[0]
+    sys.stderr.write(f"{message=}")
+
+    assert json.loads(message) == {
+        "id": "test",
+        "added_keys": ["testprefix/k1", "testprefix/k2", "testprefix/k3"],
+        "updated_keys": ["testprefix/k4"],
+        "deleted_keys": ["testprefix/k5"],
+    }
+
+
+@mock_aws
+def test_output_file_action_treats_invalid_message_json_as_permanent_error():
+    class TestMessager(Messager[str]):
+        def process_msg(self, msg: str) -> Sequence[Action]:
+            return (Messager.OutputFileAction(file_body=b"test_body", cat_path="k"),)
+
+        def gen_empty_catalogue_message(self, msg: str) -> dict:
+            return {"id": sys.stderr}
+
+    conn = boto3.resource("s3")
+    client = boto3.client("s3")
+    conn.create_bucket(Bucket="testbucket")
+
+    producer = Mock()
+
+    testmessager = TestMessager(client, "testbucket", "testprefix/", producer)
+    assert testmessager.consume("") == Messager.Failures(permanent=True, temporary=False)
+
+
+@mock_aws
+def test_output_file_action_treats_pulsar_timeout_as_temporary_error():
+    class TestMessager(Messager[str]):
+        def process_msg(self, msg: str) -> Sequence[Action]:
+            return (Messager.OutputFileAction(file_body=b"test_body", cat_path="k"),)
+
+        def gen_empty_catalogue_message(self, msg: str) -> dict:
+            return {"id": "test"}
+
+    conn = boto3.resource("s3")
+    client = boto3.client("s3")
+    conn.create_bucket(Bucket="testbucket")
```

Comment:
Do we need to specify the `CreateBucketConfiguration` parameter here defining the `LocationConstraint` key? Some tests seem to fail without this being included and set to the current AWS location, e.g. `eu-west-2`